bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -211,6 +212,28 @@ public synchronized boolean add(DiscoveryResult result) throws IllegalStateExcep
         return false;
     }
 
+    private Thing checkIfThingAlreadyExists(DiscoveryResult result) {
+        Thing thing = this.thingRegistry.get(result.getThingUID());
+        if (thing != null) {
+            // thing with same UID is already in registry
+            return thing;
+        }
+
+        // check, if there is a thing with the same representation property
+        Object value = result.getRepresentationPropertyValue();
+        if (value != null) {
+            Collection<Thing> things = this.thingRegistry.getAll();
```

Comment:
This could be expressed much nicer with `thingRegistry.stream().map(...).filter(...)`!